crates/dbsp/src/trace/layers/column_layer/mod.rs

```diff
@@ -144,6 +157,17 @@ impl<K, R> ColumnLayer<K, R> {
         F: FnMut(&K, &R) -> bool,
     {
         let original_len = self.len();
+
+        let lower_bound = self.lower_bound;
+        self.lower_bound = 0;
+
+        // SAFETY: We initialize BackshiftOnDrop with `lower_bound` offset below,
+        // so these elements will never get touched again.
+        unsafe {
+            ptr::drop_in_place(&mut self.keys[0..lower_bound]);
+            ptr::drop_in_place(&mut self.diffs[0..lower_bound]);
+        }
+
         // Avoid double drop if the drop guard is not executed,
         // since we may make some holes during the process.
         unsafe { self.set_len(0) };
```

Comment:
```suggestion
        // SAFETY: We initialize BackshiftOnDrop with `lower_bound` offset below,
        // so these elements will never get touched again.
        unsafe {
            // Avoid double drop if the drop guard is not executed,
            // or if dropping elements up to `lower_bound` panics
            // since we may make some holes during the process.
            self.set_len(0);
        
            // Drop all keys up to `lower_bound`
            let keys = ptr::slice_from_raw_parts_mut(self.keys.as_mut_ptr(), lower_bound);
            ptr::drop_in_place(keys);
            
            // Drop all diffs up to `lower_bound`
            let diffs = ptr::slice_from_raw_parts_mut(self.diffs.as_mut_ptr(), lower_bound);
            ptr::drop_in_place(diffs);
        }
```

Reply:
what was the problem with my code?

Reply:
If a panic occurred while dropping either of the slices then the drop implementations of `keys` and `diffs` would still run, causing a double free on the elements `0..lower_bound` within them